Mathlib/LinearAlgebra/Multilinear/DirectSum.lean

```diff
@@ -0,0 +1,162 @@
+        ∪ @Fintype.piFinset _ myinst _ _ (fun j ↦ (update x i b j).support))
+        (fun p _ hp ↦ by
+          letI := myinst
+          exact fromDirectSum_aux1 _ _ f _ hp)]
+    rw [Finset.sum_congr rfl (fun p _ ↦ by rw [fromDirectSum_aux2 _ _ _ p (a + b)])]
+    erw [Finset.sum_congr rfl (fun p _ ↦ (f p).map_add _ i (a (p i)) (b (p i)))]
+    rw [Finset.sum_add_distrib]
+    congr 1
+    · conv_lhs => rw [← Finset.sum_congr rfl (fun p _ ↦ by rw [← fromDirectSum_aux2 _ _ _ p a]),
+                    Finset.sum_congr (Finset.union_assoc _ _ _) (fun _ _ ↦ rfl)]
+      rw [Finset.sum_union_eq_left (fun p _ hp ↦ by
+        letI := myinst
+        exact fromDirectSum_aux1 _ _ f _ hp)]
+    · conv_lhs => rw [← Finset.sum_congr rfl (fun p _ ↦ by rw [← fromDirectSum_aux2 _ _ _ p b]),
+        Finset.sum_congr (Finset.union_assoc _ _ _) (fun _ _ ↦ rfl),
+        Finset.sum_congr (Finset.union_comm _ _) (fun _ _ ↦ rfl),
+        Finset.sum_congr (Finset.union_assoc _ _ _) (fun _ _ ↦ rfl)]
+      rw [Finset.sum_union_eq_left (fun p _ hp ↦ by
+        letI := myinst
+        exact fromDirectSum_aux1 _ _ f _ hp)]
+  map_smul' x i c a := by
+    simp only
+    rename_i myinst _ _ _ _ _ _ _
+    conv_lhs => rw [← Finset.sum_union_eq_right (s₁ := @Fintype.piFinset _ myinst _ _
+      (fun j ↦ (update x i a j).support))
+        (fun p _ hp ↦ by
+          letI := myinst
+          exact fromDirectSum_aux1 _ _ f _ hp),
+      Finset.sum_congr rfl (fun p _ ↦ by rw [fromDirectSum_aux2 _ _ _ p _])]
+    erw [Finset.sum_congr rfl (fun p _ ↦ (f p).map_smul _ i c (a (p i)))]
+    rw [← Finset.smul_sum]
+    conv_lhs => rw [← Finset.sum_congr rfl (fun p _ ↦ by rw [← fromDirectSum_aux2 _ _ _ p _]),
+      Finset.sum_union_eq_left (fun p _ hp ↦ by
+          letI := myinst
+          exact fromDirectSum_aux1 _ _ f _ hp)]
+
+@[simp]
+theorem fromDirectSum_apply (f : Π (p : Π i, κ i), MultilinearMap R (fun i ↦ M i (p i)) M')
+    (x : Π i, ⨁ (j : κ i), M i j) : fromDirectSum R κ f x =
+    ∑ p in Fintype.piFinset (fun i ↦ (x i).support), f p (fun i ↦ x i (p i)) := rfl
+
+/-- The construction `MultilinearMap.fromDirectSum`, as an `R`-linear map.-/
+def fromDirectSumₗ : ((p : Π i, κ i) → MultilinearMap R (fun i ↦ M i (p i)) M') →ₗ[R]
+    MultilinearMap R (fun i ↦ ⨁ j : κ i, M i j) M' where
+  toFun := fromDirectSum R κ
+  map_add' f g := by
+    ext x
+    simp only [fromDirectSum_apply, Pi.add_apply, MultilinearMap.add_apply]
+    rw [Finset.sum_add_distrib]
+  map_smul' c f := by
+    ext x
+    simp only [fromDirectSum_apply, Pi.smul_apply, MultilinearMap.smul_apply, RingHom.id_apply]
+    rw [Finset.smul_sum]
+
+@[simp]
+theorem fromDirectSumₗ_apply (f : Π (p : Π i, κ i), MultilinearMap R (fun i ↦ M i (p i)) M')
+    (x : Π i, ⨁ (j : κ i), M i j) : fromDirectSumₗ R κ f x =
+    ∑ p in Fintype.piFinset (fun i ↦ (x i).support), f p (fun i ↦ x i (p i)) := rfl
+
+theorem _root_.piFinset_support_lof_sub (p : Π i, κ i) (a : Π i, M i (p i)) :
```

Comment:
Should this be in the `Fintype` namespace?

Reply:
I deleted this lemma: really it's just `support_lof` / `support_of`, and it wasn't needed after my refactor.